notification-service/src/test/java/com/redhat/parodos/notification/service/NotificationRecordServiceTests.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (c) 2022 Red Hat Developer
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.parodos.notification.service;
+
+import com.redhat.parodos.notification.enums.Operation;
+import com.redhat.parodos.notification.enums.State;
+import com.redhat.parodos.notification.jpa.entity.NotificationMessage;
+import com.redhat.parodos.notification.jpa.entity.NotificationRecord;
+import com.redhat.parodos.notification.jpa.entity.NotificationUser;
+import com.redhat.parodos.notification.jpa.repository.NotificationRecordRepository;
+import com.redhat.parodos.notification.jpa.repository.NotificationUserRepository;
+import com.redhat.parodos.notification.service.impl.NotificationRecordServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+
+/**
+ * @author Nir Argaman (Github: nirarg)
+ */
+
+@Transactional
+@ActiveProfiles("test")
+public class NotificationRecordServiceTests {
+
+	private NotificationRecordRepository notificationRecordRepository;
+
+	private NotificationUserRepository notificationUserRepository;
+
+	private NotificationRecordService notificationRecordServiceImpl;
+
+	@BeforeEach
+	void initNotificationRecordServiceImpl() {
+		this.notificationRecordRepository = Mockito.mock(NotificationRecordRepository.class);
+		this.notificationUserRepository = Mockito.mock(NotificationUserRepository.class);
+		this.notificationRecordServiceImpl = new NotificationRecordServiceImpl(notificationRecordRepository,
+				notificationUserRepository);
+	}
+
+	@Test
+	void createNotificationRecords() {
+		NotificationMessage notificationMessage = Mockito.mock(NotificationMessage.class);
+		NotificationUser notificationUser = Mockito.mock(NotificationUser.class);
+		List<NotificationUser> records = Arrays.asList(notificationUser);
```

Comment:
In such situation, I'd like to recommend to use `Collections.singletonList` instead of `Arrays.asList`. In this way, `notificationUser` will be Immutable. Since we're testing, this will increase the “control” on the object and on the array as well.

Reply:
Thanks, changed all relevant places